### PR TITLE
Generics

### PR DESCRIPTION
**Pull request by asiajanicka**

### CODE REVIEW

File: src/main/java/org/jjm/hotel/Hotel.java [hunk 1]
```diff
@@ -53,6 +53,18 @@ public Room getFirstAvailableRoom() throws NoPlacementAvailableException {
...
 
+    public Room getRoomByType(RoomType roomType) throws NoPlacementAvailableException {
+        List<Room> roomsOfGivenType = rooms.stream()
+                .filter(p -> !p.isBooked() && p.getType().equals(roomType))
+                .collect(Collectors.toList());
+        if (roomsOfGivenType.size() == 0) {
+            throw new NoPlacementAvailableException(
+                    String.format("There is no free room of type %s in the hotel %s. All rooms are booked.",
+                            roomType, name));
+        } else
```
Comment (vboltrukanis) [inline src/main/java/org/jjm/hotel/Hotel.java]: even if there is only one line in else, still use brackets { }

File: src/main/java/org/jjm/hotel/Hotel.java [hunk 2]
```diff
@@ -53,6 +53,18 @@ public Room getFirstAvailableRoom() throws NoPlacementAvailableException {
             return getAllAvailableRooms().get(0);
     }
 
+    public Room getRoomByType(RoomType roomType) throws NoPlacementAvailableException {
+        List<Room> roomsOfGivenType = rooms.stream()
+                .filter(p -> !p.isBooked() && p.getType().equals(roomType))
+                .collect(Collectors.toList());
+        if (roomsOfGivenType.size() == 0) {
```
Comment (vboltrukanis) [inline src/main/java/org/jjm/hotel/Hotel.java]: in method name we don't specify that we want an empty room, maybe we just wanted to check if such romm exists in this hotel.

File: src/main/java/org/jjm/transport/Flight.java [hunk 3]
```diff
@@ -1,28 +1,37 @@
...
+    public Seat<PlaneSeatType> getSeatByType(PlaneSeatType seatType) throws NoPlacementAvailableException {
+        return Utils.findSeatByType(getSeats(), seatType);
+    }
+
     @Override
     public String toString() {
         return String.format("Flight: %s Boarding time: %s",
                 super.toString(), DateFormat.format(getBoardingTime()));
     }
+
```
Comment (vboltrukanis) [inline src/main/java/org/jjm/transport/Flight.java]: no need for this line

File: src/main/java/org/jjm/transport/Seat.java [hunk 4]
```diff
@@ -7,17 +7,19 @@
 import java.math.BigDecimal;
 import java.util.Objects;
 
-public class Seat implements IBook {
+public class Seat <T> implements IBook {
```
Comment (vboltrukanis) [inline src/main/java/org/jjm/transport/Seat.java]: This is too general, you better specify it like `public class Seat <T extends String> implements` or whatever seat type should be.

**Review by vboltrukanis — APPROVED** [review]
Minor fixes, will not re-review this. Feel free to to merge as soon as previous PRs are merged and this one is fixed according to your opinion.

File: src/main/java/org/jjm/hotel/Hotel.java (hunk 1 above)
Comment (asiajanicka) [reply src/main/java/org/jjm/hotel/Hotel.java]: fixed

File: src/main/java/org/jjm/hotel/Hotel.java (hunk 2 above)
Comment (asiajanicka) [reply src/main/java/org/jjm/hotel/Hotel.java]: fixed - removed checking if !p.isBooked() is true

File: src/main/java/org/jjm/transport/Flight.java (hunk 3 above)
Comment (asiajanicka) [reply src/main/java/org/jjm/transport/Flight.java]: fixed

File: src/main/java/org/jjm/transport/Seat.java (hunk 4 above)
Comment (asiajanicka) [reply src/main/java/org/jjm/transport/Seat.java]: changed to Seat <T extends Enum>

**Pull request merged**